libraries/chain/steem_evaluator.cpp

```diff
@@ -316,6 +316,35 @@ void delete_comment_evaluator::do_apply( const delete_comment_operation& o )
    _db.remove( comment );
 }
 
+struct comment_options_extension_visitor
+{
+   comment_options_extension_visitor( const comment_object& c, database& db ) : _c( c ), _db( db ) {}
+
+   typedef void result_type;
+
+   const comment_object& _c;
+   database& _db;
+
+   void operator()( const comment_payout_beneficiaries& cpb ) const
```

Comment:
require that beneficiares are all unique and sorted by weight then account name, ideally this check would be done as part of static validation.